### PR TITLE
`d/shared_image_version`: fix `id`

### LINKED ISSUE #17307: azurerm_shared_image_version ID not returning version
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.2.3

### AzureRM Provider Version

3.10.0

### Affected Resource(s)/Data Source(s)

azurerm_shared_image_version

### Terraform Configuration Files

```hcl
data "azurerm_shared_image_version" "main" {
  gallery_name        = "SIG"
  image_name          = "Win10"
  name                = "latest"
  resource_group_name = "sig"
}

output "id" "main" {
  value = data.azurerm_shared_image_version.main.id
}
```

### Debug Output/Panic Output

```shell
id = /subscriptions/{subscription-id}/resourceGroups/sig/providers/Microsoft.Compute/galleries/Prod/images/Win10/versions/latest
```

### Expected Behaviour

Return Azure Resource ID with version (Was working - Seems more of an Azure API problem but worth asking here) i.e

/subscriptions/{subscription-id}/resourceGroups/sig/providers/Microsoft.Compute/galleries/Prod/images/Win10/versions/<ins>**1.0.0**</ins>

### Actual Behaviour

Returns

/subscriptions/{subscription-id}/resourceGroups/sig/providers/Microsoft.Compute/galleries/Prod/images/Win10/versions/<ins>**latest**</ins>

### Steps to Reproduce

terraform apply

### Important Factoids

_No response_

### References

_No response_

### PR DESCRIPTION
**Pull request by myc2h6o**
Fix #17307. id seems to be set to `latest` by mistake in [this change](https://github.com/hashicorp/terraform-provider-azurerm/pull/14934/files#diff-12e064c48daacb4c1b9cb364b8c84324c957c44bf2ea72af2d0f41ab3dc9dcf6), updating it with the exact version name retrieved from the image version.

### CODE REVIEW

**Comment by myc2h6o** [comment]
Test result
image

File: internal/services/compute/shared_image_version_data_source.go [hunk 1]
```diff
@@ -118,7 +118,8 @@ func dataSourceSharedImageVersionRead(d *pluginsdk.ResourceData, meta interface{
 		return err
 	}
 
-	d.SetId(id.ID())
+	exactId := parse.NewSharedImageVersionID(subscriptionId, id.ResourceGroup, id.GalleryName, id.ImageName, *image.Name)
```
Comment (mbfrahry) [inline internal/services/compute/shared_image_version_data_source.go:121]: We need to confirm that `*image.Name` isn't `nil` before casting to it

**Review by mbfrahry — CHANGES_REQUESTED** [review]
Hey @myc2h6o, this looks good but we do have to do a nil check so we don't crash

**Comment by myc2h6o** [comment]
Hi @mbfrahry thanks for reviewing the change! I've added the nil validation for image.Name, could you take another look?

**Comment by myc2h6o** [comment]
Updated test result
image

**Review by katbyte — APPROVED** [review]
LGTM 🌻

**Pull request merged**